src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Serialization/JsonReaderExtensions.cs

```diff
@@ -54,5 +53,68 @@ public static string ReadNextStringProperty(this JsonReader reader, string prope
 
             throw new JsonSerializationException($"Could not find string property '{propertyName}'.");
         }
+
+        public static (int? hash, string propertyValue) ReadPropertyAndHash(this JsonReader reader, string propertyName)
```

Comment:
Feels odd  to have  this on the generic JsonReaderExtensions. Seems  more like a helper method in  the TagHelperDescriptor deserializer (not everything  serializes the hashcode)